tests/unit/test_data.py

```diff
@@ -0,0 +1,119 @@
+import globus_sdk
+
+from tests.framework import (CapturedIOTestCase, get_client_data,
+                             SDKTESTER1A_NATIVE1_RT,
+                             GO_EP1_ID, GO_EP2_ID)
+
+
+class DataTests(CapturedIOTestCase):
+
+    def setUp(self):
+        """
+        Creates two TransferData objects and two Delete objects for testing,
+        one with defaults and one with custom paramaters
+
+        """
+        super(DataTests, self).setUp()
+
+        ac = globus_sdk.NativeAppAuthClient(
+            client_id=get_client_data()["native_app_client1"]["id"])
+        authorizer = globus_sdk.RefreshTokenAuthorizer(
+            SDKTESTER1A_NATIVE1_RT, ac)
+        self.tc = globus_sdk.TransferClient(authorizer=authorizer)
+
+        # constants
+        self.label = "label"
+        self.params = {"param1": "value1", "param2": "value2"}
+
+        # default transfer data
+        self.tdata = globus_sdk.TransferData(self.tc, GO_EP1_ID, GO_EP2_ID)
+        # transfer data with paramaters
+        self.param_tdata = globus_sdk.TransferData(self.tc,
+                                                   GO_EP1_ID, GO_EP2_ID,
+                                                   label=self.label,
+                                                   sync_level="exists",
+                                                   **self.params)
+        # default delete data
+        self.ddata = globus_sdk.DeleteData(self.tc, GO_EP1_ID)
+        # delete data with paramaters
+        self.param_ddata = globus_sdk.DeleteData(self.tc, GO_EP1_ID,
+                                                 label=self.label,
+                                                 recursive="True",
+                                                 **self.params)
+
+    def test_tranfer_init(self):
+        """
+        Verifies TransferData field initialization
+        """
+
+        # default init
+        self.assertEqual(self.tdata["DATA_TYPE"], "transfer")
+        self.assertEqual(self.tdata["source_endpoint"], GO_EP1_ID)
+        self.assertEqual(self.tdata["destination_endpoint"], GO_EP2_ID)
+        self.assertIn("submission_id", self.tdata)
+        self.assertIn("DATA", self.tdata)
+        self.assertEqual(len(self.tdata["DATA"]), 0)
```

Comment:
Doing initialization steps in `setUp()` and then testing the results in your test functions is a little unusual.
It's not that bad, but it will pose a problem if we ever want to test out various initialization behaviors.

My preference would be to move the creation of `TransferData` and `DeleteData` to helpers, with the ability to pass overrides for various specific values. i.e. allow callers to `self.make_tdata()` or `self.make_tdata(sync_level="checksum")`

It's not necessary for the current set of tests, but it may be in the future.